### PR TITLE
Rabbitmq Weak Credentials Plugin

### PR DESCRIPTION
**Pull request by leonardo-doyensec**
Created a plugin for testing weak credentials in rabbitmq management portal

### CODE REVIEW

File: google/detectors/credentials/generic_weak_credential_detector/src/test/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTesterTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  private static final TestCredential WEAK_CRED_1 =
+      TestCredential.create("user", Optional.of("1234"));
+  private static final TestCredential WEAK_CRED_2 =
+      TestCredential.create("root", Optional.of("pass"));
+  private static final TestCredential WRONG_CRED_1 =
+      TestCredential.create("wrong", Optional.of("pass"));
+
+  private static final String WEAK_CRED_AUTH_1 = "Basic dXNlcjoxMjM0";
+  private static final String WEAK_CRED_AUTH_2 = "Basic cm9vdDpwYXNz";
+  private static final ServiceContext.Builder rabbitmqServiceContext =
```
Comment (dawidg-doyen) [inline google/detectors/credentials/generic_weak_credential_detector/src/test/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTesterTest.java:61]: use capitals if it's static

File: google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+          .append("http://")
+          .append(NetworkEndpointUtils.toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(path);
+    return targetUrlBuilder.toString();
+  }
+
+  // Checks if the response body contains elements of a rabbitmq management page - custom
+  // fingerprinting phase
```
Comment (dawidg-doyen) [inline google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java:84]: Inconsistent use of `fingerprinting`, use `fingerprint` form as you used down below.

File: google/detectors/credentials/generic_weak_credential_detector/src/test/resources/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/testdata/successfulAuthdResponse.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"name":"","tags":["administrator"]}
```
Comment (dawidg-doyen) [inline google/detectors/credentials/generic_weak_credential_detector/src/test/resources/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/testdata/successfulAuthdResponse.json:1]: should the name be empty or should it contain name i.e. 'admin' ?

**Review by dawidg-doyen — CHANGES_REQUESTED** [review]
Done with the review. Rest of the comments provided via chat. Remember to run the checkstyle linter at the end.

**Comment by leonardo-doyensec** [comment]
Updated everything as you requested. Regarding the empty username in the json file i've replaced it with a placeholder that at run-time will be updated with the current username

**Pull request closed without merging**